### PR TITLE
Enabled Ubuntu 18.04 for Open MPI and MVAPICH MPI

### PR DESCRIPTION
**Pull request by juhlee-microsoft**
Enabled Open MPI in Ubuntu 18.04
Enabled MVAPICH MPI in Ubuntu 18.04
Fixed indentation in working files

### CODE REVIEW

**Comment by juhlee-microsoft** [comment]
### OPEN MPI
PS C:\Users\juhlee\LISAv2> .\Run-LisaV2.ps1 -TestPlatform Azure -TestLocation "southcentralus" -RGIdentifier "juhlee" -ARMImageName "Canonical UbuntuServer 18.04-LTS latest" -TestNames "INFINIBAND-OPEN-MPI-2VM" -XMLSecretFile <secret file>

[LISAv2 Test Results Summary]
Test Run On           : 03/11/2019 23:44:22
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:29

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 INFINIBAND-OPEN-MPI-2VM                                                           PASS                20.85
        InfiniBand-Verification-1-FirstBoot : eth0 IP : PASS
        InfiniBand-Verification-1-FirstBoot : IBV_PINGPONG : PASS
        InfiniBand-Verification-1-FirstBoot : PingPong Intranode : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-MPI1 : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-RMA : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-NBC : PASS
        InfiniBand-Verification-2-Reboot : eth0 IP : PASS
        InfiniBand-Verification-2-Reboot : IBV_PINGPONG : PASS
        InfiniBand-Verification-2-Reboot : PingPong Intranode : PASS
        InfiniBand-Verification-2-Reboot : IMB-MPI1 : PASS
        InfiniBand-Verification-2-Reboot : IMB-RMA : PASS
        InfiniBand-Verification-2-Reboot : IMB-NBC : PASS


Logs can be found at C:\Users\juhlee\LISAv2\TestResults\2019-11-03-16-44-15-6169


03/12/2019 00:14:01 : [INFO ] Current Git Branch:  juhlee/rdma_ub1804
03/12/2019 00:14:01 : [INFO ] Current Git Commit ID: fb8452df2119989795fa7130d89c4019085bd68f
03/12/2019 00:14:01 : [INFO ] Creating 'C:\Users\juhlee\LISAv2\Azure-CY29-TestLogs.zip' from 'C:\Users\juhlee\LISAv2\TestResults\2019-11-03-16-44-15-6169'
03/12/2019 00:14:01 : [INFO ] C:\Users\juhlee\LISAv2\Azure-CY29-TestLogs.zip created successfully.
03/12/2019 00:14:01 : [INFO ] Analyzing results..
03/12/2019 00:14:01 : [INFO ] LISAv2 exit code: 0


### MVAPICH MPI

PS C:\Users\juhlee\LISAv2> .\Run-LisaV2.ps1 -TestPlatform Azure -TestLocation "southcentralus" -RGIdentifier "juhlee" -ARMImageName "Canonical UbuntuServer 18.04-LTS latest" -TestNames "INFINIBAND-MVAPICH-MPI-2VM" -XMLSecretFile <secret file>

[LISAv2 Test Results Summary]
Test Run On           : 03/12/2019 00:20:42
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:29

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 INFINIBAND-MVAPICH-MPI-2VM                                                        PASS                18.96
        InfiniBand-Verification-1-FirstBoot : eth0 IP : PASS
        InfiniBand-Verification-1-FirstBoot : IBV_PINGPONG : PASS
        InfiniBand-Verification-1-FirstBoot : PingPong Intranode : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-MPI1 : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-RMA : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-NBC : PASS
        InfiniBand-Verification-2-Reboot : eth0 IP : PASS
        InfiniBand-Verification-2-Reboot : IBV_PINGPONG : PASS
        InfiniBand-Verification-2-Reboot : PingPong Intranode : PASS
        InfiniBand-Verification-2-Reboot : IMB-MPI1 : PASS
        InfiniBand-Verification-2-Reboot : IMB-RMA : PASS
        InfiniBand-Verification-2-Reboot : IMB-NBC : PASS


Logs can be found at C:\Users\juhlee\LISAv2\TestResults\2019-11-03-17-20-39-1728


03/12/2019 00:50:16 : [INFO ] Current Git Branch:  juhlee/rdma_ub1804
03/12/2019 00:50:16 : [INFO ] Current Git Commit ID: fb8452df2119989795fa7130d89c4019085bd68f
03/12/2019 00:50:16 : [INFO ] Creating 'C:\Users\juhlee\LISAv2\Azure-TN90-TestLogs.zip' from 'C:\Users\juhlee\LISAv2\TestResults\2019-11-03-17-20-39-1728'
03/12/2019 00:50:16 : [INFO ] C:\Users\juhlee\LISAv2\Azure-TN90-TestLogs.zip created successfully.
03/12/2019 00:50:16 : [INFO ] Analyzing results..
03/12/2019 00:50:16 : [INFO ] LISAv2 exit code: 0

**Review by lizzha — DISMISSED** [review]

File: Testscripts/Windows/VERIFY-INFINIBAND-MultiVM.ps1 [hunk 1]
```diff
@@ -1,424 +1,423 @@
...
 
-        # Reboot VM to apply RDMA changes
-        Write-LogInfo "Rebooting All VMs!"
-        $TestProvider.RestartAllDeployments($AllVMData)
+		# Reboot VM to apply RDMA changes
+		Write-LogInfo "Rebooting All VMs!"
+		$TestProvider.RestartAllDeployments($AllVMData)
 
-        # In some cases, IB will not be initialized after reboot
-        Resolve-UninitializedIB
```
Comment (adriansuhov) [inline Testscripts/Windows/VERIFY-INFINIBAND-MultiVM.ps1:172]: this function was introduced for SLES testing - on 32 VMs test cases some of the VMs didn't initialize the IB after the first reboot. But after a secondary reboot, all was good. We need to run this function at least for SLES use case

Comment (juhlee-microsoft) [reply Testscripts/Windows/VERIFY-INFINIBAND-MultiVM.ps1:172]: Will make sure of SLES15 and SLES 12 SP4 testing.

Comment (iamshital) [reply Testscripts/Windows/VERIFY-INFINIBAND-MultiVM.ps1:172]: > this function was introduced for SLES testing - on 32 VMs test cases some of the VMs didn't initialize the IB after the first reboot. But after a secondary reboot, all was good. We need to run this function at least for SLES use case

@adriansuhov , Do we have an active bug for this issue in TFS?

Comment (juhlee-microsoft) [reply Testscripts/Windows/VERIFY-INFINIBAND-MultiVM.ps1:172]: @iamshital , @adriansuhov is working with internal teams for further investigation. Will find more data soon.

Comment (juhlee-microsoft) [reply Testscripts/Windows/VERIFY-INFINIBAND-MultiVM.ps1:172]: Fixed in 33d1e401d9b470b70444c43d69ac962292bad790

**Comment by juhlee-microsoft** [comment]
Added back kernel module check, and found master branch code broken in RHEL7.5/SLES15.
Will open new PR for bug fix later.

### Test result

[LISAv2 Test Results Summary]
Test Run On           : 03/11/2019 23:44:22
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:29

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 INFINIBAND-OPEN-MPI-2VM                                                           PASS                20.85 
	InfiniBand-Verification-1-FirstBoot : eth0 IP : PASS 
	InfiniBand-Verification-1-FirstBoot : IBV_PINGPONG : PASS 
	InfiniBand-Verification-1-FirstBoot : PingPong Intranode : PASS 
	InfiniBand-Verification-1-FirstBoot : IMB-MPI1 : PASS 
	InfiniBand-Verification-1-FirstBoot : IMB-RMA : PASS 
	InfiniBand-Verification-1-FirstBoot : IMB-NBC : PASS 
	InfiniBand-Verification-2-Reboot : eth0 IP : PASS 
	InfiniBand-Verification-2-Reboot : IBV_PINGPONG : PASS 
	InfiniBand-Verification-2-Reboot : PingPong Intranode : PASS 
	InfiniBand-Verification-2-Reboot : IMB-MPI1 : PASS 
	InfiniBand-Verification-2-Reboot : IMB-RMA : PASS 
	InfiniBand-Verification-2-Reboot : IMB-NBC : PASS 


Logs can be found at C:\Users\juhlee\LISAv2\TestResults\2019-11-03-16-44-15-6169


03/12/2019 00:14:01 : [INFO ] Current Git Branch:  juhlee/rdma_ub1804
03/12/2019 00:14:01 : [INFO ] Current Git Commit ID: fb8452df2119989795fa7130d89c4019085bd68f
03/12/2019 00:14:01 : [INFO ] Creating 'C:\Users\juhlee\LISAv2\Azure-CY29-TestLogs.zip' from 'C:\Users\juhlee\LISAv2\TestResults\2019-11-03-16-44-15-6169'
03/12/2019 00:14:01 : [INFO ] C:\Users\juhlee\LISAv2\Azure-CY29-TestLogs.zip created successfully.
03/12/2019 00:14:01 : [INFO ] Analyzing results..
03/12/2019 00:14:01 : [INFO ] LISAv2 exit code: 0

**Review by lizzha — APPROVED** [review]

**Pull request merged**